src/app/store/user-metadata-atoms.ts

```diff
@@ -0,0 +1,18 @@
+import { atom } from 'jotai'
+import { atomWithStorage } from 'jotai/utils'
+
+import { GetApiV2UserAddress200 } from '@/gen'
+
+export const userMetadataAtom = atomWithStorage<GetApiV2UserAddress200 | null>(
+  'userMetadata',
+  null,
+)
+
+export const fetchUserAtom = atom(null, async (_, set, address: string) => {
+  try {
+    const user = await (await fetch(`/api/user/${address}`)).json()
+    set(userMetadataAtom, user)
```

Comment:
couldn't you just use the address here (to avoid the fetch) - as the user basically doesn't hold much else?

Reply:
yeah of course we could, I can turn this off for now altogether, but this is basically the userMetadata, so I just added it because it does no harm, and when we store settings or anything like a name we can ask of the user, we have the system in place.